packages/template-chakra-storefront/src/hooks/use-wish-list.js

```diff
@@ -35,16 +61,137 @@ export const useWishList = ({listId = ''} = {}) => {
             parameters: {customerId},
-            // we only use one type of product lists for now
             body: {type: 'wish_list'}
         })
     }, [productLists, isProductListsSuccess])
 
     const wishLists = productLists?.data?.filter((list) => list.type === 'wish_list') || []
     const currentWishlist = wishLists.find((list) => list.id === listId)
+    const wishlist = !listId ? wishLists[0] : currentWishlist
+
+    const isItemInWishlist = useCallback(
+        (product, variant) => {
+            if (!wishlist || !product) {
+                return false
+            }
+            const productId = normalizeProductId(product, variant)
+            return !!wishlist.customerProductListItems?.find((item) => item.productId === productId)
+        },
+        [wishlist]
+    )
+
+    const addToWishlist = useCallback(
+        async (product, variant, quantity = 1) => {
+            try {
+                if (!customerId || !wishlist) {
+                    return
+                }
+
+                const productId = normalizeProductId(product, variant)
+
+                if (isItemInWishlist(product, variant)) {
+                    toast({
+                        title: formatMessage(TOAST_MESSAGE_ALREADY_IN_WISHLIST),
+                        type: 'info',
+                        action: (
+                            <Button variant="link" onClick={() => navigate('/account/wishlist')}>
+                                {formatMessage(TOAST_ACTION_VIEW_WISHLIST)}
+                            </Button>
+                        )
+                    })
+                    return
+                }
+
+                await createCustomerProductListItem.mutateAsync({
+                    parameters: {
+                        listId: wishlist.id,
+                        customerId
+                    },
+                    body: {
+                        quantity,
+                        productId,
+                        public: false,
+                        priority: 1,
+                        type: 'product'
+                    }
+                })
+
+                toast({
+                    title: formatMessage(TOAST_MESSAGE_ADDED_TO_WISHLIST, {quantity: 1}),
+                    type: 'success',
+                    action: (
+                        <Button variant="link" onClick={() => navigate('/account/wishlist')}>
+                            {formatMessage(TOAST_ACTION_VIEW_WISHLIST)}
+                        </Button>
+                    )
+                })
+            } catch (error) {
+                console.error(error)
+                toast({
+                    title: formatMessage(API_ERROR_MESSAGE),
+                    type: 'error'
+                })
+            }
+        },
+        [
+            customerId,
+            wishlist,
+            isItemInWishlist,
+            createCustomerProductListItem,
+            toast,
+            formatMessage,
+            navigate
```

Comment:
Why would be want to re-assign the `addToWishlist` variable when formatMessage changes? Will that never change? Same goes for toast and navigate, maybe createCustomerProductListItem too?

Reply:
cleaned up the dependency array